File: ResourceManager/Models/Configurations/ActivityTagConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class ActivityTagConfiguration : IEntityTypeConfiguration<ActivityTag>
    {
        public void Configure(EntityTypeBuilder<ActivityTag> builder)
        {
            builder.ToTable("ActivityTags");
            builder.HasKey(x => new { x.ActivityID, x.TagID });

            builder.HasOne(x => x.Activity)
                .WithMany(x => x.ActivityTags)
                .HasForeignKey(x => x.ActivityID);

            builder.HasOne(x => x.Tag)
                .WithMany(x => x.ActivityTags)
                .HasForeignKey(x => x.TagID);

        }
    }
}

File: ResourceManager/Models/Configurations/ActivityTagConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore

Namespace ResourceManager.Models.Configurations
    Public Class ActivityTagConfiguration
        Implements IEntityTypeConfiguration(Of ActivityTag)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of ActivityTag)) Implements IEntityTypeConfiguration(Of ActivityTag).Configure
            builder.ToTable("ActivityTags")
            builder.HasKey(Function(x) New With {x.ActivityID, x.TagID
            })

            builder.HasOne(Function(x) x.Activity).WithMany(Function(x) x.ActivityTags).HasForeignKey(Function(x) x.ActivityID)

            builder.HasOne(Function(x) x.Tag).WithMany(Function(x) x.ActivityTags).HasForeignKey(Function(x) x.TagID)

        End Sub
    End Class
End Namespace
